Sources/Rego/Builtins/Casts.swift

```diff
@@ -0,0 +1,41 @@
+import AST
+import Foundation
+
+extension BuiltinFuncs {
+    static func toNumber(ctx: BuiltinContext, args: [AST.RegoValue]) async throws -> AST.RegoValue {
+        guard args.count == 1 else {
+            throw BuiltinError.argumentCountMismatch(got: args.count, want: 1)
+        }
+        switch args[0] {
+        case .boolean(let v):
+            return v ? 1 : 0
+        case .null:
+            return 0
+        case .number(_):
+            return args[0]
+        case .string(let s):
+            // For compatibility with Go, we do not allow leading pluses
+            guard !s.hasPrefix("+") else {
```

Comment:
Capturing observations:

There's something more subtle going on.

* In Go, `+123` is a halting error, while `++123` is regular error (`undefined` in non-strict mode). It appears the parsing error is occurring at different layers (json vs. ParseFloat)
* In WASM, `+123` is _not_ a halting error, just a regular error.
* In both WASM and Go, parsing `123.0` will evaluate as `123.0` rather than `123` as in the Swift version. Not sure whether that is considered diverging behavior, as the two are equivalent. 🤔

Who wins, Go or WASM?